<a href="https://colab.research.google.com/github/gaso/tipologia_heart-attack-analysis/blob/main/analysis.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

<div style="width: 100%; clear: both;">
<div style="float: left; width: 50%;">
<img src="http://www.uoc.edu/portal/_resources/common/imatges/marca_UOC/UOC_Masterbrand.jpg", align="left">
</div>
<div style="float: right; width: 50%;">
<p style="margin: 0; padding-top: 22px; text-align:right;">Tipología y ciclo de vida de los datos</p>
<p style="margin: 0; text-align:right;">2023-1 · Máster universitario en Ciencia de datos (Data science)</p>
<p style="margin: 0; text-align:right; padding-button: 100px;">Estudios de Informática, Multimedia y Telecomunicación</p>
</div>
</div>
<div style="width:100%;">&nbsp;</div>

Cargue de Librerias

In [ ]:
install.packages('ggplot2')

Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)



# 1. Descripción del dataset.

## 1.1 Carga del dataset

In [5]:
path= 'heart.csv'
data <- read.csv(path, row.names=NULL)

str(data)

'data.frame':	303 obs. of  14 variables:
 $ age     : int  63 37 41 56 57 57 56 44 52 57 ...
 $ sex     : int  1 1 0 1 0 1 0 1 1 1 ...
 $ cp      : int  3 2 1 1 0 0 1 1 2 2 ...
 $ trtbps  : int  145 130 130 120 120 140 140 120 172 150 ...
 $ chol    : int  233 250 204 236 354 192 294 263 199 168 ...
 $ fbs     : int  1 0 0 0 0 0 0 0 1 0 ...
 $ restecg : int  0 1 0 1 1 1 0 1 1 1 ...
 $ thalachh: int  150 187 172 178 163 148 153 173 162 174 ...
 $ exng    : int  0 0 0 0 1 0 0 0 0 0 ...
 $ oldpeak : num  2.3 3.5 1.4 0.8 0.6 0.4 1.3 0 0.5 1.6 ...
 $ slp     : int  0 0 2 2 2 1 1 2 2 2 ...
 $ caa     : int  0 0 0 0 0 0 0 0 0 0 ...
 $ thall   : int  1 2 2 2 2 1 2 3 3 2 ...
 $ output  : int  1 1 1 1 1 1 1 1 1 1 ...


El conjunto de datos contiene **303** observaciones y **14** variables. 

**age:** La edad del paciente (variable numérica).

**sex:** El sexo del paciente (1: masculino, 0: femenino) (variable numérica).

**cp:** Tipo de dolor en el pecho (variable numérica).

**trtbps:** Presión arterial en reposo (variable numérica).

**chol:** Nivel de colesterol en mg/dL (variable numérica).

**fbs:** Nivel de azúcar en la sangre en ayunas (1: superior a 120 mg/dL, 0: inferior a 120 mg/dL) (variable numérica).

**restecg:** Resultados del electrocardiograma en reposo (variable numérica).

**thalachh:** Frecuencia cardíaca máxima alcanzada (variable numérica).

**exng:** Angina inducida por ejercicio (1: sí, 0: no) (variable numérica).

**oldpeak:** Depresión del ST inducida por el ejercicio en relación con el descanso (variable numérica).

**slp: **Pendiente del segmento ST de ejercicio (variable numérica).

**caa:** Número de vasos principales coloreados por fluoroscopia (variable numérica).

**thall:** Resultados de las pruebas de estrés con talio (variable numérica).

**output:** Variable de salida, 1 si la persona tiene enfermedad cardíaca y 0 si no la tiene (variable numérica).